app/helpers/application_helper.rb

```diff
@@ -10,4 +10,19 @@ def service_name
   def l(date, options = {})
     super(date, **options) if date
   end
+
+  # NOTE: implicit decorators assumed to be in app/decorators
+  def decorate(object, decorator_class = nil)
+    decorator_class ||= "#{object.class.to_s.demodulize}Decorator".constantize
+    decorator = decorator_class.new(object, self)
+    yield(decorator) if block_given?
```

Comment:
Will this work without the ampersand parameter?

Reply:
the `&` is used to pass a reference to a named block to another method usually. `yield(thing)` passes the `thing` to the calling block.

```
decorate(thing) do |decorated_thing_yielded_here|
  decorated_thing_yielded_here.my_decorator_method
end
```